### PR TITLE
Implemented logic to read CURRENCY and NULL_FLAGS field types

### PR DESCRIPTION
**Pull request by jigneshkhatri**
Implemented logic to read CURRENCY ('Y') and NULL_FLAGS ('0') field types.

### CODE REVIEW

**Comment by jigneshkhatri** [comment]
@yortus Can you please review and merge this pull request? I need to read CURRENCY and NULL_FLAGS from DBF files in one of my projects.

File: src/dbf-file.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import * as assert from 'assert';
 import * as iconv from 'iconv-lite';
-import {extname} from 'path';
-import {FieldDescriptor, validateFieldDescriptor} from './field-descriptor';
-import {isValidFileVersion} from './file-version';
-import {CreateOptions, Encoding, normaliseCreateOptions, normaliseOpenOptions, OpenOptions} from './options';
-import {close, open, read, stat, write} from './utils';
-import {createDate, format8CharDate, formatVfpDateTime, parseVfpDateTime, parse8CharDate} from './utils';
+import { extname } from 'path';
```
Comment (yortus) [inline src/dbf-file.ts:3]: can you please remove all the whitespace changes from the PR and only change the code where you are adding new functionality?

**Review by yortus — CHANGES_REQUESTED** [review]
Thanks for the PR @jigneshkhatri. To merge and release this, it will need a few additions:
- support `Y` and `0` field types when _writing_ dbf files
- add validation for `Y` and `0` field types (in `field-descriptor.ts`)
- add test fixture(s) that has `Y` and `0` field types
- add unit tests that successfully read and write files with `Y` and `0` fields

File: src/dbf-file.ts (hunk 1 above)
Comment (jigneshkhatri) [reply src/dbf-file.ts:3]: test_cases

Do you know why complied test files are created under `test/test` directory? Ideally they should be created under `test` directory.

Comment (jigneshkhatri) [reply src/dbf-file.ts:3]: I am just running command `npm run build` to compile the source code.

Comment (yortus) [reply src/dbf-file.ts:3]: I'm not sure how you are getting that. I can't reproduce the above screenshot either locally or using codespaces. Running `npm install` then `npm run build` creates the built test files under `dist/test` as per the `tsconfig` and npm script.

**Comment by yortus** [comment]
Closing due to inactivity. Feel free to re-open to address the [review feedback](https://github.com/yortus/DBFFile/pull/60#pullrequestreview-817118787).

**Pull request closed without merging**